src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,513 @@
+            }
+        }
+        return task;
+    }
+
+    public static String convertTodotoLine(ToDo task) {
+        if (task.isDone()) {
+            return ("T | √ | " + task.getDescription());
+        } else {
+            return ("T | X | " + task.getDescription());
+        }
+    }
+
+    public static String convertDeadlinetoLine(Deadline task) {
+        if (task.isDone()) {
+            return ("D | √ | " + task.getDescription() + " | " + task.getBy());
+        } else {
+            return ("D | X | " + task.getDescription() + " | " + task.getBy());
+        }
+    }
+
+    public static String convertEventtoLine(Event task) {
+        if (task.isDone()) {
+            return ("E | √ | " + task.getDescription() + " | " + task.getAt());
+        } else {
+            return ("E | X | " + task.getDescription() + " | " + task.getAt());
+        }
+    }
+
+    public static void initializeList(ArrayList<Task> tasks) throws FileNotFoundException {
+        File f = new File(LIST_STORAGE_FILE);
+        Scanner s = new Scanner(f);
+        while (s.hasNext()) {
+            tasks.add(convertLinetoTask(s.nextLine()));
+        }
+        printList(tasks, tasks.size());
+    }
+
+    public static void initializeTaskLineList(ArrayList<String> taskLines) throws FileNotFoundException {
+        File f = new File(LIST_STORAGE_FILE);
+        Scanner s = new Scanner(f);
+        while (s.hasNext()) {
+            taskLines.add(s.nextLine());
+        }
+    }
+
+    public static void appendToFile(String textToAdd) throws IOException {
+        FileWriter fw = new FileWriter(LIST_STORAGE_FILE, true);
+        fw.write(textToAdd);
+        fw.close();
+    }
+
+    public static void clearFile() throws IOException {
+        FileWriter fw = new FileWriter(LIST_STORAGE_FILE);
+        fw.write("");
+        fw.close();
+    }
+
+    public static void storetoHardDisk(ArrayList<String> taskLines) {
+        try {
+            clearFile();
+        } catch (IOException e) {
+            System.out.println("Something went wrong: " + e.getMessage());
+        }
+        for (String taskLine : taskLines) {
+            try {
+                appendToFile(taskLine + System.lineSeparator());
+            } catch (IOException e) {
+                System.out.println("Something went wrong: " + e.getMessage());
+            }
+        }
+    }
+
+    public static void main(String[] args){
+        printGreeting();
+        initializeHardDiskSave();
+        ArrayList<Task> tasks = new ArrayList<>();
+        ArrayList<String> taskLines = new ArrayList<>();
+        try {
+            initializeTaskLineList(taskLines);
+            initializeList(tasks);
+        } catch (FileNotFoundException e) {
+            System.out.println("File not found, pls try again");
+        }
+
+        Scanner in = new Scanner(System.in);
+        while (true) {
+            String command = in.nextLine();
+            if (command.equals("bye")) {
```

Comment:
String literals like `"bye", "echo"` are considered as constants so perhaps you can store them as constants for better readabililty